static/app/utils/replays/getDiffTimestamps.tsx

```diff
@@ -0,0 +1,56 @@
+import type {Event} from 'sentry/types/event';
+import type ReplayReader from 'sentry/utils/replays/replayReader';
+import type {ReplayFrame} from 'sentry/utils/replays/types';
+
+export function getReplayDiffOffsetsFromFrame(
+  replay: ReplayReader | null,
+  frame: ReplayFrame
+) {
+  return {
+    leftOffsetMs: frame.offsetMs,
+    rightOffsetMs:
+      (frame.data.mutations.next?.timestamp ?? 0) -
+      (replay?.getReplay().started_at.getTime() ?? 0),
+  };
+}
+
+export function getReplayDiffOffsetsFromEvent(replay: ReplayReader, event: Event) {
+  const startTimestampMS =
+    'startTimestamp' in event ? event.startTimestamp * 1000 : undefined;
+  const timeOfEvent = event.dateCreated ?? startTimestampMS ?? event.dateReceived;
+  const eventTimestampMs = timeOfEvent ? Math.floor(new Date(timeOfEvent).getTime()) : 0;
+  // `event.dateCreated` is the most common date to use, and it's in seconds not ms
+
+  const hydrationFrame = replay
+    .getChapterFrames()
+    .find(
+      breadcrumb =>
+        'category' in breadcrumb &&
+        breadcrumb.category === 'replay.hydrate-error' &&
+        breadcrumb.timestampMs > eventTimestampMs &&
+        breadcrumb.timestampMs < eventTimestampMs + 1000
+    );
+
+  if (hydrationFrame) {
+    return getReplayDiffOffsetsFromFrame(replay, hydrationFrame);
+  }
+
+  const frames = replay.getRRWebFrames();
+  const replayStartTimestamp = replay?.getReplay().started_at.getTime() ?? 0;
+
+  const leftReplayFrameIndex = replay
+    .getRRWebFrames()
+    .findIndex(frame => frame.timestamp < eventTimestampMs);
+  const leftFrame = frames.at(Math.max(0, leftReplayFrameIndex));
+  const leftOffsetMs = replayStartTimestamp - (leftFrame?.timestamp ?? 0);
```

Comment:
am i misunderstanding, or is `replayStartTimestamp` smaller than `leftFrame.timestamp`?  so this could be negative?

Reply:
(or is it being negative expected)

Reply:
I set some Math.max() in here, so we'll never get a negative number back.

For the left side, a `0` offset is probably ok to deal with. We could have either an empty html page (unlikely, because we need some html to load the sdk!) or it's just a really early snapshot. 

If the right side is `0` then it's a problem for the diff views... I think in that case those views should deal with that themselves... for example we could show an error to the user. `0` in that case might mean that there's no 2nd snapshot to compare against.